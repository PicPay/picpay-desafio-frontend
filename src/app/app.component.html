<!-- <app-user-row></app-user-row>
<app-modal></app-modal> -->
<app-home></app-home>