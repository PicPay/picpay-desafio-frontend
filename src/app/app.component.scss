@import '~src/sass/variables';

.content {
  display: flex;
  height: 100%;

  & .page {
    flex: 1;
    overflow: auto;
    background-color: $base-color;
  }
  & .payments {
    width: 400px;
    overflow: auto;
  }
}