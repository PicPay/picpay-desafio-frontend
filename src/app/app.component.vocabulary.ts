export const APP_VOCABULARY = {
  header: {
    title: 'app.header.title',
    themeSwitcher: 'app.header.themeSwitcher',
    changeTheme: 'app.header.changeTheme',
    userFilter: {
      label: 'app.header.userFilter.label',
      all: 'app.header.userFilter.all',
      paid: 'app.header.userFilter.paid',
      pending: 'app.header.userFilter.pending',
    },
  },
  snackBar: {
    close: 'app.snackBar.close',
  },
};
