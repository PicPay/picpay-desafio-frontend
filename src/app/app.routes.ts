import { Routes } from "@angular/router";

export const appRoutes: Routes = [
    {
        path: 'usuario',
        loadChildren: () => import('./modules/usuario/usuario.module').then(m => m.UsuarioModule)
    },
    {
        path: '**',
        redirectTo: 'usuario',
        pathMatch: 'full'
    }
]