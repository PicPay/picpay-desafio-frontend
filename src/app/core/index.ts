export * from './components/index';
export * from './model/index';
export * from './services/index';
