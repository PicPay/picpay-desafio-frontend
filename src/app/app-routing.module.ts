import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  { path: '', redirectTo: '/pagar', pathMatch: 'full' },
  { path: 'pagar', loadChildren: './pages/payment/payment.module#PaymentModule' },
  { path: '404', redirectTo: '' },
  { path: '**', redirectTo: '' }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}
