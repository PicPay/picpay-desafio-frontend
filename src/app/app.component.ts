import { Component, OnInit } from '@angular/core';
import { User } from 'src/models/user';
import { AppService } from './app.service';

@Component({
    selector: 'app-root',
    templateUrl: './app.component.html',
    styleUrls: ['./app.component.scss'],
    providers: [AppService],
})
export class AppComponent implements OnInit {
    title = 'Desafio Picpay Front-end';
    users: User[];

    constructor(private appService: AppService) {}

    ngOnInit() {
        this.getUsers();
    }

    getUsers() {
        this.appService.getUsers().subscribe((ResponseUsers: User[]) => {
            this.users = ResponseUsers;
        });
    }

    openModal(user: User) {
        this.appService.openModal(user);
    }
}
