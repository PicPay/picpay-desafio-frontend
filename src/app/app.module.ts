import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { UserListComponent } from './components/user-list/user-list.component';
import { TransactionModalComponent } from './components/modal/transaction-modal/transaction-modal.component';
import { ReciptModalComponent } from './components/modal/recipt-modal/recipt-modal.component';
import { AppRoutingModule } from './app-routing.module';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CurrencyMaskModule } from 'ng2-currency-mask';
import { LOCALE_ID } from '@angular/core';
import localePt from '@angular/common/locales/pt';
import {registerLocaleData} from '@angular/common';
registerLocaleData(localePt);

@NgModule({
  declarations: [
    AppComponent,
    UserListComponent,
    TransactionModalComponent,
    ReciptModalComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    ReactiveFormsModule,
    FormsModule,
    CurrencyMaskModule
  ],
  providers: [{
    provide: LOCALE_ID, 
    useValue: "pt-BR"
  }],
  bootstrap: [AppComponent]
})
export class AppModule { }
