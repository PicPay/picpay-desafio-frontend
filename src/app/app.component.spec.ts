import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [AppComponent],
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const el = fixture.debugElement.nativeElement.outerHTML;
    expect(el).toMatchSnapshot();
  });
});
