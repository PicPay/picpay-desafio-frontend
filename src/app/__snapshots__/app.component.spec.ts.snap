// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snapshot 1`] = `"<div id=\\"root1\\" ng-version=\\"8.2.14\\"><app-header></app-header><router-outlet></router-outlet></div>"`;
